### DISCUSSION TITLE
Pagination in Blazor

### DISCUSSION THREAD

**Discussion by ashishsinha24** [MarimerLLC/csla#2352] [Questions]
What is an appropriate way to implement pagination in CSLA.
I don't want to sprinkle code all over my business classes. There should be an elegant way to handle this.

How is everyone else handling this?

**Comment by rockfordlhotka** [comment]
Your list class(es) do need to support being fetched a page at a time, there's no real way to avoid this.

There was (just last week?) a discussion thread here on this exact topic - how to implement per-page fetch operations in a list class. I answered with some code to show how it is done.

You can do both btw - have a complete fetch and a per-page fetch - in the same list class.

**Reply by rockfordlhotka** [reply]
https://github.com/MarimerLLC/csla/discussions/2337

**Reply by ashishsinha24** [reply]
Thanks, I missed that. Even couldnt find it in search
